github-page/_docs/decorators/prop.md

```diff
@@ -70,6 +70,8 @@ class Defaulted {
 }
 ```
 
+Note that if you don't specify the property as optional (i.e. with the question mark) you will be forced to specify the property at creation time. 
```

Comment:
is this still so with #249 merged?

Reply:
Yes, this is an unfortunate side-effect of being unable to check whether a property is `required: true` from TypeScript because we can't get decorator type data. However, I added an extra type parameter called `ExtraPartialKeys` that allows circumventing this.

```ts
class Test {
    @prop({ required: true, default: 'Bob' })
    firstName: string;
    @prop({ required: true })
    lastName: string;
    set fullName(name: string) {  }
    get fullName() { return this.firstName + ' ' + this.lastName }
}
const TestModel = getModelForClass(Test);
TestModel.create<'fullName', 'type'>({});
```

Reply:
i already saw, and actually hoped it would get in there